src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -56,11 +96,175 @@ public static void DisposeAndClearHandle(ref SafeCurlMultiHandle curlHandle)
                 }
             }
 
+            internal void PollFds(List<libc.pollfd> readyFds)
+            {
+                int count;
+                libc.pollfd[] pollFds;
+
+                readyFds.Clear();
+
+                lock (this)
+                {
+                    // TODO: Avoid the allocation when count is in 100s
+                    count = _fdSet.Count + 1;
+                    pollFds = new libc.pollfd[count];
+
+                    // Always include special fd in the poll set
```

Comment:
Consider augmenting the comment to explain the purpose of the special fd, e.g. ". This fd is signaled when there are any changes to the _fdSet, allowing us to cancel and restart the poll."